tests/test_prepush_cmd.py

```diff
@@ -0,0 +1,247 @@
+        """
+        GIVEN a prepush range with 0 commits
+        WHEN the command is run
+        THEN it should return 0 and warn it was unable to get range
+        """
+        get_list_mock.return_value = []
+        result = cli_fs_runner.invoke(
+            cli,
+            ["-v", "scan", "pre-push"],
+            env={"PRE_COMMIT_FROM_REF": "a" * 40, "PRE_COMMIT_TO_REF": "b" * 40},
+        )
+        assert result.exit_code == 0
+        assert "Unable to get commit range." in result.output
+
+    @patch("ggshield.hook_cmd.get_list_commit_SHA")
+    def test_prepush_too_many(self, get_list_mock: Mock, cli_fs_runner: CliRunner):
+        """
+        GIVEN a prepush range with 51 commits
+        WHEN the command is run
+        THEN it should return 0 warn too many commits for scanning, and scan last 50
+        """
+        get_list_mock.return_value = ["a"] * 51
+        result = cli_fs_runner.invoke(
+            cli,
+            ["-v", "scan", "pre-push"],
+            env={"PRE_COMMIT_FROM_REF": "a" * 40, "PRE_COMMIT_TO_REF": "b" * 40},
+        )
+        assert result.exit_code == 0
+        assert "Too many commits. Scanning last 50" in result.output
+
+    @patch("ggshield.hook_cmd.get_list_commit_SHA")
+    @patch("ggshield.hook_cmd.scan_commit_range")
+    @patch("ggshield.hook_cmd.check_git_dir")
+    def test_prepush_pre_commit_framework_new(
+        self,
+        check_dir_mock: Mock,
+        scan_commit_range_mock: Mock,
+        get_list_mock: Mock,
+        cli_fs_runner: CliRunner,
+    ):
+        """
+        GIVEN a prepush range with a 20 commits through the new env vars of the framework
+        WHEN the command is run
+        THEN it should pass onto scan and return 0
+        """
+        scan_commit_range_mock.return_value = 0
+        commit_list = ["a" for _ in range(20)]
+        get_list_mock.return_value = commit_list
+
+        result = cli_fs_runner.invoke(
+            cli,
+            ["-v", "scan", "pre-push"],
+            env={"PRE_COMMIT_FROM_REF": "a" * 40, "PRE_COMMIT_TO_REF": "b" * 40},
+        )
+        get_list_mock.assert_called_once_with(
+            "--max-count=51 " + "b" * 40 + "..." + "a" * 40
+        )
+        scan_commit_range_mock.assert_called_once_with(
+            client=ANY,
+            cache=ANY,
+            commit_list=commit_list,
+            output_handler=ANY,
+            verbose=True,
+            filter_set=set(),
+            matches_ignore=ANY,
+            all_policies=False,
+            scan_id=ANY,
+            mode_header="pre_push",
+            banlisted_detectors=set(),
+        )
+        assert "Commits to scan: 20" in result.output
+        assert result.exit_code == 0
+
+    @patch("ggshield.hook_cmd.get_list_commit_SHA")
+    @patch("ggshield.hook_cmd.scan_commit_range")
+    @patch("ggshield.hook_cmd.check_git_dir")
+    def test_prepush_pre_commit_framework_old(
```

Comment:
minor: seems it may be merged into the previous one with a parameterize test
(minor meaning do what you want with this comment)